src/Parser/RestoDruid/Modules/Traits/EternalRestoration.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+import {HOTS_AFFECTED_BY_ESSENCE_OF_GHANIR} from '../../Constants';
+
+const ETERNAL_RESTORATION_INCREASE = 1;
+const EOG_BASE_DURATION = 8;
+/**
+ * Eternal Restoration (Artifact Trait)
+ * Increases the duration of Essence of G'Hanir by 1 sec
+ */
+class EternalRestoration extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  rank = 0;
+  healing = 0;
+  lastEoGApplied = null;
+  eogDuration = 0;
+
+  on_initialized() {
+    this.rank = this.combatants.selected.traitsBySpellId[SPELLS.ETERNAL_RESTORATION.id];
+    this.active = this.rank > 0;
+    this.eogDuration = ((this.rank * ETERNAL_RESTORATION_INCREASE)+EOG_BASE_DURATION)*1000;
+  }
+
+  on_byPlayer_heal(event) {
+    const spellId = event.ability.guid;
+
+    // We are interested in the healing done on the last second of EoG
+    if(this.lastEoGApplied != null
+      && (this.lastEoGApplied + this.eogDuration - 1000) <= event.timestamp
+      && this.combatants.selected.hasBuff(SPELLS.ESSENCE_OF_GHANIR.id)
+      && HOTS_AFFECTED_BY_ESSENCE_OF_GHANIR.indexOf(spellId) !== -1) {
+      if(SPELLS.REGROWTH.id === spellId && event.tick !== true) {
```

Comment:
Technically you don't need to check for Regrowth here, could just check that it's not a tick. Also, `event.tick !== true` should be `!event.tick`.

Reply:
Done